### DISCUSSION TITLE
Github Actions Workflow Environment Variables Secrets

### DISCUSSION THREAD

**Discussion by mcabrejos24** [community/community#25247] [Actions]
<p>I’ve search everywhere and I still can’t find something that works for me.</p>
<p>I am trying to access the github environment secrets I set for my repository, but each time I reference it, I get an empty value back.</p>
<p>Below you can see that I’m calling the<code> secrets.DELETE</code> key in the <code>env:</code> and that I"m also trying to set it below but at neither point is the environment variable <code>DELETE</code> set.</p>
<pre><code class="lang-auto">name: Django CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
  workflow_dispatch:
    inputs:
      logLevel:
        description: 'Log level'     
        required: true
        default: 'warning' 
        type: choice
        options:
        - info
        - warning
        - debug 
      tags:
        description: 'Test scenario tags'
        required: false 
        type: boolean
      environment:
        description: 'Environment to run tests against'
        type: environment
        required: true 

env:
  DELETE: ${{ secrets.delete }}

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      max-parallel: 4
      matrix:
        python-version: [3.7, 3.8, 3.9]

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install Dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt
    - name: Lint with Flake8
      run: |
        flake8 .
    - name: mess around
      env:
        DELETE: ${{ secrets.delete }}
      run: echo "$DELETE"
    - name: Run Tests
      run: |
        python manage.py test

</code></pre>
<p>The secrets are set in my repository.</p>
<p>I’m not sure what I’m missing.</p>

**Comment by jsoref** [comment, marked as answer]
<p>You might try:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181089845-43424c6f-aa9b-4667-bcec-325e2b93413f.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging" target="_blank" rel="noopener">Enabling debug logging - GitHub Docs</a></h3>

  <p>If the workflow logs do not provide enough detail to diagnose why a workflow, job, or step is not working as expected, you can enable additional debug logging.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by mcabrejos24** [comment]
<p>Wow okay after reading this I went to go add the <code>ACTIONS_RUNNER_DEBUG</code> secret key and realized I had put my secret keys under the environment secret keys instead of repository. After changing it, it worked!</p>
<p>Thanks a lot!</p>
